Source/Core/Core/Scripting/Python/Pyr.h

```diff
@@ -0,0 +1,197 @@
+#pragma once
+
+#include <Python.h>
+#include "Common/Logging/Log.h"
+
+namespace Pyr
+{
+class Object
+{
+  PyObject* m_py_object_;
+  Object(PyObject* py_object) : m_py_object_(py_object) {}
+
+public:
+  Object& operator=(Object&&) = default;
+  Object& operator=(const Object& rhs)
+  {
+    Py_XINCREF(rhs.unwrap());
+    Py_XDECREF(m_py_object_);
+    m_py_object_ = rhs.unwrap();
+    return *this;
+  }
+  Object(Object&&) = default;
+  Object(Object& rhs)
+  {
+    m_py_object_ = rhs.unwrap();
+    Py_XINCREF(rhs.unwrap());
+  }
+
+  ~Object() { Py_XDECREF(m_py_object_); }
+  PyObject* unwrap() const { return m_py_object_; }
+  bool is_null() const { return m_py_object_ == NULL; }
+  static Object wrap_existing(PyObject* py_object) { return Object(py_object); }
+};
+
+inline Object wrap(PyObject* py_object)
+{
+  return Object::wrap_existing(py_object);
+}
+inline Object take(PyObject* py_object)
+{
+  Py_XINCREF(py_object);
+  return Object::wrap_existing(py_object);
+}
+static Object Null = wrap(NULL);
+
+template <typename T>
+static constexpr PyObject* convert_to_py(T n)
+{
+  // using the constexpr checks is easier than template overloading imo
+  if constexpr (std::is_same<T, bool>::value)
+  {
+    return PyBool_FromLong(n);
+  }
+  else if constexpr (std::is_floating_point<T>::value)
+  {
+    return PyFloat_FromDouble(n);
+  }
+  else if constexpr (std::is_integral<T>::value)
+  {
+    static_assert(sizeof(T) <= sizeof(unsigned long long), "integral type too big, not supported");
+    const bool is_signed = std::is_signed<T>::value;
+    if (sizeof(T) <= sizeof(unsigned long))
+      return is_signed ? PyLong_FromLong(n) : PyLong_FromUnsignedLong(n);
+    return is_signed ? PyLong_FromLongLong(n) : PyLong_FromUnsignedLongLong(n);
+  }
+  else
+  {
+    static_assert(false, "Type (see callstack) cannot be converted to Python type");
+  }
+}
+
+template <typename T>
+static constexpr const char* get_pyformat()
+{
+  if constexpr (std::is_same<T, signed char>::value)
```

Comment:
These can all just use our type aliases s8, s16, etc where applicable. Also, function call syntax can be used here.

Reply:
I chose to write the types out like that to mirror the documentation on https://docs.python.org/3/c-api/arg.html#numbers . Is that acceptable? I'll add a documentation link for clarification